pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -88,61 +86,61 @@ func (r *MigPlan) GetAssetCollection(client k8sclient.Client) (*MigAssetCollecti
 	return GetAssetCollection(client, r.Spec.MigAssetCollectionRef)
 }
 
-// PlanRefResources holds structs for the resources MigPlans reference.
-// This data is passed to functions within MigStage and MigMigration reconcile.
-type PlanRefResources struct {
+// Resources referenced by the plan.
+// Contains all of the fetched referenced resources.
+type PlanResources struct {
 	MigPlan        *MigPlan
 	MigAssets      *MigAssetCollection
 	MigStorage     *MigStorage
 	SrcMigCluster  *MigCluster
 	DestMigCluster *MigCluster
-
-	SrcBackup   *velerov1.Backup
-	DestRestore *velerov1.Restore
 }
 
-// GetRefResources gets referenced resources from a MigPlan and loads them into a ReferencedResources struct
-func (r *MigPlan) GetRefResources(client k8sclient.Client, logPrefix string) (*PlanRefResources, error) {
-	resources := &PlanRefResources{}
-
-	// MigPlan
-	resources.MigPlan = r
-
+// GetRefResources gets referenced resources from a MigPlan.
+func (r *MigPlan) GetRefResources(client k8sclient.Client) (*PlanResources, error) {
 	// MigAssetCollection
 	migAssets, err := r.GetAssetCollection(client)
 	if err != nil {
-		log.Info(fmt.Sprintf("[%s] Failed to GET MigAssetCollection referenced by MigPlan [%s/%s]",
-			logPrefix, r.Namespace, r.Name))
 		return nil, err
 	}
-	resources.MigAssets = migAssets
+	if migAssets == nil {
+		return nil, errors.New("asset-collection not found")
+	}
 
 	// MigStorage
 	storage, err := r.GetStorage(client)
 	if err != nil {
-		log.Info(fmt.Sprintf("[%s] Failed to GET MigAssetCollection referenced by MigStorage [%s/%s]",
-			logPrefix, r.Namespace, r.Name))
 		return nil, err
 	}
-	resources.MigStorage = storage
+	if storage == nil {
+		return nil, errors.New("storage not found")
+	}
 
 	// SrcMigCluster
 	srcMigCluster, err := r.GetSourceCluster(client)
 	if err != nil {
-		log.Info(fmt.Sprintf("[%s] Failed to GET SrcMigCluster referenced by MigPlan [%s/%s]",
-			logPrefix, r.Namespace, r.Name))
 		return nil, err
 	}
-	resources.SrcMigCluster = srcMigCluster
+	if srcMigCluster == nil {
+		return nil, errors.New("source cluster not found")
+	}
 
 	// DestMigCluster
 	destMigCluster, err := r.GetDestinationCluster(client)
 	if err != nil {
-		log.Info(fmt.Sprintf("[%s] Failed to GET DestMigCluster referenced by MigPlan [%s/%s]",
-			logPrefix, r.Namespace, r.Name))
 		return nil, err
 	}
-	resources.DestMigCluster = destMigCluster
+	if destMigCluster == nil {
+		return nil, errors.New("destination cluster not found")
+	}
+
+	resources := &PlanResources{
```

Comment:
+1, think this makes it clearer what is being loaded into the `PlanResources` struct for someone reading this function for the first time.